src/boot/linux.c

```diff
@@ -119,16 +46,63 @@ EFI_STATUS linux_exec(
         if (err != EFI_SUCCESS)
                 return log_error_status(err, "Bad kernel image: %m");
 
-        _cleanup_(unload_imagep) EFI_HANDLE kernel_image = NULL;
-        err = load_image(parent, kernel->iov_base, kernel->iov_len, &kernel_image);
```

Comment:
ie: retain this, move the new loading into a different function, and call the new function either if this fails as a fallback, or if shim_loaded() returns false, either should be fine

Reply:
@valentindavid above comment still unaddressed

Reply:
I am trying to address it, but it is not straight forward. And it is taking me some time to test. But I am hoping to get a solution that pleases everybody.

Reply:
That sounds perfectly fine to me, thanks!

Reply:
Any chance the check for secure_boot_enabled could be dropped, please? The reason is that I cannot ask you to verify these changes on anything other than x86-64 qemu as it would be way too onerous, and given the thing with relocations, I am afraid of regressions on hardware platforms like armv7, riscv, loongarch64. So I'd prefer if, in case shim is used as the first stage, we go through it even with secureboot disabled, as we don't have that enabled outside of amd64/arm64, so that I can tell people opening bugs downstream to just use shim as the first stage.

Reply:
> Are you talking about automated tests for regression? Or doing manual tests? I am not sure what is onerous here.

Manual tests - you really have access and are willing to spend time to try this on real hardware armv7/riscv/loong boards? If so, that is positively impressive :-)
usually we never require such level of testing outside x86, as it's hard to have all that actual hw available, and it's very time-consuming

I mean this is not super important, I would just prefer to have the same workflows happen regardless of the firmware configuration, and instead just depend on what's actually installed on the ESP, as I have a feeling that would make the whole process more predictable and hence my job easier, that's all